builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -144,7 +148,10 @@ func resourceAwsS3BucketObjectPut(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("Error putting object in S3 bucket (%s): %s", bucket, err)
 	}
 
-	d.Set("etag", resp.ETag)
+	// See https://forums.aws.amazon.com/thread.jspa?threadID=44003
+	d.Set("etag", strings.Trim(*resp.ETag, `"`))
+
+	d.Set("version_id", resp.VersionId)
 	d.SetId(key)
 	return nil
```

Comment:
why doesn't the Put call the Read? I know you didn't change this part but seems like it should